tiledb/api/src/filter/mod.rs

```diff
@@ -223,24 +223,13 @@ impl FilterData {
                 }
                 | CompressionType::DoubleDelta {
                     reinterpret_datatype,
-                } => {
-                    // these filters do not accept floating point
-                    let check_type =
-                        if let Some(Datatype::Any) = reinterpret_datatype {
-                            *input
-                        } else if let Some(reinterpret_datatype) =
-                            reinterpret_datatype
-                        {
-                            reinterpret_datatype
-                        } else {
-                            return None;
-                        };
-                    if check_type.is_real_type() {
-                        None
+                } => reinterpret_datatype.map_or(Some(*input), |dtype| {
+                    if !dtype.is_real_type() {
+                        Some(dtype)
```

Comment:
This looks to me like a regression if `dtype == Datatype::Any`, shouldn't there still be a case for that?

Or, as a philosophical matter, should we remove `Datatype::Any` from the Rust layer and change the places where it is allowed to `Option<Datatype>`?  Or should we draw inspiration from `NonZero<T>` and make a `DatatypeNotAny`?

But none of these are really right since `Any` is an actual type which requires `CellVarNum::Var`.

Reply:
Well crap. I apparently managed t delete all the proptests code I wrote this weekend. Not a big deal as it was mostly just a learning exercise, though I did have a quite nice filter list generation property. I'll work later on getting that rewritten to show off.

I'll have to double check whether or not passing Any is accepted as a reinterpret_datatype or if it gets tossed in an exception. It could go both ways at multiple layers. If it does then yeah, this code is rejecting a different corner of the truth table instead.

Reply:
Proptests stuff rewritten and on a branch (so I don't do another dumb and accidentally delete it). You definitely caught an edge case there. If a user sets the datatype to Any we have to handle it because TileDB accepts that.

PR is updated with the fix.